docs/source/relayer.rst

```diff
@@ -0,0 +1,105 @@
+  The agent utilizes these proofs in an optimistic way. If a challenge game ever occurs, or if a
+  fill was invalid, the agent has the chance to prove that a misbehavior occured and in the case of a challenge game,
+  get rewarded for it.  
+|
+| Since these **proofs** lay on a different chain than the chain where the requests originated from and where the challenge
+  game is running, we need a way to transport this **proof** from one chain to another in order to be able 
+  to prove that a certain state is correct. Thats where the cross-chain relayer comes into play.
+|
+| For instructions on how to setup, compile and run the relayer, make sure you take a look at the `/relayer/README.md` file.
+
+Command reference
+-----------------
+
+The relayer CLI currently supports the following commands:
+
+
+* :ref:`command-relay` relays a fill proof or a fill invalidation message from chain A to chain B.
+* :ref:`command-prove-op-message` proves on L1 that a message exists on L2. Used only for messages traveling from chains that are based on the *OP Bedrock* stack.
+
+.. _command-relay:
+
+``relay``
+^^^^^^^^^^^^^^^^
+
+The ``relay`` command takes care of relaying a message from one chain to another (E2E).
+In cases where a message needs to be relayed from an "OP Bedrock"-like chain, then one first needs to run
+the :ref:`command-prove-op-message` command, for the same message that needs to be relayed, before he can run the :ref:`command-relay` command.
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - Command-line option 
+     - Description
+
+   * - ``--l1-rpc-url URL``
+     - RPC URL to be used for communicating with the base chain.
+
+   * - ``--l2-relay-to-rpc-url URL``
+     - RPC URL to be used for communicating with the destination chain (chain where the proof is travelling *to*).
+
+   * - ``--l2-relay-from-rpc-url URL``
+     - RPC URL to be used for communicating with the source chain (chain where the proof is travelling *from*).
+
+   * - ``--keystore-file PATH``
+     - Path to the account keyfile.
+
+   * - ``--password PASSWORD``
+     - Password of the keystore file.
+
+   * - ``--l2-transaction-hash TX_HASH``
+     - Transaction hash of the submitted message that needs to be relayed.
+
+   * - ``--network-from PATH``
+     - Path to a file with custom network configuration. This option is mainly used for development purposes.
+
+   * - ``--network-to PATH``
+     - Path to a file with custom network configuration. This option is mainly used for development purposes.
+
+
+.. _command-prove-op-message:
+
+``prove-op-message``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``prove-op-message`` command is used only when a message needs to travel *from* "OP Bedrock"-like networks.
+On such networks, there is a need of proving the message upfront before it is treated as a message
+that should be included in a batch and submitted to L1 (after it reaches the finality period).
+The finality period starts counting only after the message was proven on L2.
```

Comment:
Nitpick: I would drop `counting` here (the finality period starts and ends, it does not count).